src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -1,11 +1,144 @@
+import logging
+
+from django.db import IntegrityError
+from django.utils import timezone
+
+from rest_framework import serializers, status
 from rest_framework.response import Response
 
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.serializers import serialize
-from sentry.models import ProjectCodeOwners
+from sentry.models import (
+    ProjectCodeOwners,
+    ExternalUser,
+    ExternalTeam,
+    RepositoryProjectPathConfig,
+    User,
+)
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+from sentry.ownership.grammar import (
+    parse_code_owners,
+    convert_codeowners_syntax,
+)
+
+from sentry.api.endpoints.project_ownership import ProjectOwnershipSerializer, ProjectOwnershipMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectCodeOwnerSerializer(CamelSnakeModelSerializer):
+    code_mapping_id = serializers.IntegerField(required=True)
+    raw = serializers.CharField(required=True)
+    organization_integration_id = serializers.IntegerField(required=False)
+
+    class Meta:
+        model = ProjectCodeOwners
+        fields = ["raw", "code_mapping_id", "organization_integration_id"]
+
+    def validate(self, attrs):
+        # If it already exists, set default attrs with existing values
+        if self.instance:
+            attrs = {
+                "raw": self.instance.raw,
+                "code_mapping_id": self.instance.repository_project_path_config,
+                **attrs,
```

Comment:
@NisanthanNanthakumar what about the default value for `organization_integration_id`?

Reply:
@scefali 

> We need to create a migration to remove the organization_integration column from sentry_projectcodeowners table. We already have a reference to it from repository_project_path_config foreignkey column and it is possible to update both tables independantly, increasing the chances of a drift in consistency.

I'm going to make this `Future` PR now and merge it before this PR. It'll eliminate the need to handle `organization_integration_id`.